.github/workflows/scripts/generate-weekly-report.js

```diff
@@ -0,0 +1,428 @@
+  const sponsorNeededIssues = issuesWithLabels["Sponsor Needed"].filter(issue => filterOnDateRange({ issue, sevenDaysAgo, midnightYesterday }));
+  sponsorNeededIssues.forEach(issue => {
+    stats.issuesNewSponsorNeeded.count++;
+    const { html_url: url, title, number } = issue;
+    stats.issuesNewSponsorNeeded.data.push({ url, title, number });
+  });
+  return stats
+}
+
+function generateReport({ issuesData, previousReport, componentData }) {
+  const out = [
+    `## Format`,
+    "- `{CATEGORY}: {COUNT} ({CHANGE_FROM_PREVIOUS_WEEK})`",
+    "## Issues Report",
+    '<ul>'];
+
+  // generate report for issues
+  for (const lbl of Object.keys(issuesData)) {
+    const section = [``];
+    const { count, data, title } = issuesData[lbl];
+
+    if (previousReport === null) {
+      section.push(`<li> ${title}: ${count}`);
+    } else {
+      const previousCount = previousReport.issuesData[lbl].count; 
+      section.push(`<li> ${title}: ${count} (${count - previousCount})`);
+    }
+
+    // generate summary if issues exist
+    // NOTE: the newline after <summary> is required for markdown to render correctly
+    if (data.length !== 0) {
+      section.push(`<details>
+    <summary> Issues </summary>\n`);
+    section.push(`${data.map((issue) => {
+        const { url, title, number } = issue;
+        return `- [ ] ${title} ([#${number}](${url}))`;
+      }).join('\n')}
+</details>`);
+    }
+
+    section.push('</li>');
+    out.push(section.join('\n'));
+  }
+  out.push('</ul>');
+
+  // generate report for components
+  out.push('\n## Components Report', '<ul>');
+  for (const lbl of Object.keys(componentData)) {
+    const section = [``];
+    const data = componentData[lbl];
+    const count = Object.keys(data).length;
+
+    section.push(`<li> ${lbl}: ${count}`);
+    if (data.length !== 0) {
+    // NOTE: the newline after <summary> is required for markdown to render correctly
+      section.push(`<details>
+    <summary> Components </summary>\n`);
+      section.push(`${Object.keys(data).map((compName) => {
+        const {stability} = data[compName]
+        return `- [ ] ${compName}: ${JSON.stringify(stability)}`
+      }).join('\n')}
+</details>`);
+    }
+    section.push('</li>');
+    out.push(section.join('\n'));
+  }
+  out.push('</ul>');
+
+  // add json data
+  out.push('\n ## JSON Data');
+  out.push('<!-- MACHINE GENERATED: DO NOT EDIT -->');
+  out.push(`<details>
+<summary>Expand</summary>
+<pre>
+{
+  "issuesData": ${JSON.stringify(issuesData, null, 2)},
+  "componentData": ${JSON.stringify(componentData, null, 2)}
+}
+</pre>
+</details>`);
+  const report = out.join('\n');
+  return report;
+}
+
+async function createIssue({ github, lookbackData, report, context }) {
+  const title = `Weekly Report: ${lookbackData.sevenDaysAgo.toISOString().slice(0, 10)} - ${lookbackData.midnightYesterday.toISOString().slice(0, 10)}`;
+  return github.rest.issues.create({
+    owner: context.payload.repository.owner.login,
```

Comment:
Why do use `owner: REPO_OWNER` in some places and `owner: context.payload.repository.owner.login` in other?

Reply:
when we need to look back for old issues, we're using the hardcoded repo because that's where the actual issues are.
when we're creating a new issue, we use the github context to get the current username in order to create an issue in the forked repo (because its likely that the person that forked the repo doesn't have permissions to create an issue in the collector repo, nor should they if they are testing the script)
i've added a comment to explain this in the code